pandas/tests/indexes/multi/test_names.py

```diff
@@ -124,3 +124,12 @@ def test_get_names_from_levels():
 
     assert idx.levels[0].name == "a"
     assert idx.levels[1].name == "b"
+
+
+def test_setting_names_from_levels_raises():
+    idx = pd.MultiIndex.from_product([["a"], [1, 2]], names=["a", "b"])
+    with pytest.raises(RuntimeError, match="set_names"):
+        idx.levels[0].name = "foo"
```

Comment:
i could do something weird like:

```
lvl = idx.levels[0]

ser = pd.Series(lvl, index=lvl)
ser.index.name="foo"
```

is there a less-contrived way in which `idx.levels[0]` could come un-attached from the multiindex?

Reply:
That raises (correctly, I think). I can add a test for it if you like.